File: Fresh/UI/fresh_app/src/app/model/requests/ProductPriceUpdateRequest.ts
export class ProductPriceUpdateRequest {
    pricePerUnit: number;
    effectiveFrom: Date;
    productId: number;

    constructor(pricePerUnit: number, effectiveFrom: Date, productId: number) {
        this.pricePerUnit = pricePerUnit;
        this.effectiveFrom = effectiveFrom;
        this.productId = productId;
    }
}

File: Fresh/Fresh.Model/Requests/ProductPriceUpdateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Fresh.Model.Requests
{
    public class ProductPriceUpdateRequest
    {
        [Required(ErrorMessage = "This field is required !")]
        public decimal PricePerUnit { get; set; }

        [Required(ErrorMessage = "This field is required !")]
        public DateTime EffectiveFrom { get; set; }

        [Required(ErrorMessage = "This field is required !")]
        public int ProductId { get; set; }
    }
}
